File: 122.best-time-to-buy-and-sell-stock-ii.java
/*
 * @lc app=leetcode id=122 lang=java
 *
 * [122] Best Time to Buy and Sell Stock II
 */

// @lc code=start
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        for (int i = 0; i<prices.length; i++){
            if (i +1 == prices.length){
                break;
            }
            int currPrice = prices[i];
            int nextPrice = prices[i+1];
            if (currPrice < nextPrice) {
                profit += nextPrice - currPrice;
                continue;
            }
        }
        return profit;
    }
}
// @lc code=end

File: 122.best-time-to-buy-and-sell-stock-ii.py
#
# @lc app=leetcode id=122 lang=python3
#
# [122] Best Time to Buy and Sell Stock II
#

# @lc code=start
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        for i, v in enumerate(prices):
            if i + 1 == len(prices):
                break
            curr_price = v
            next_price =  prices[i+1]
            if curr_price < next_price:
                profit += next_price - curr_price
                continue

        return profit   

        
# @lc code=end

